test/indexer/test_hca_indexer.py

```diff
@@ -447,128 +447,177 @@ def _num_docs_by_index(self, hits) -> Mapping[Tuple[str, bool], int]:
+                                          'developmentStage=human adult stage;'
+                                          'organ=liver;'
+                                          'libraryConstructionApproach=10X v2 sequencing,Smart-seq2'
+                            },
+                            {
+                                'uuid': '7c3ad02f-2a7a-5229-bebd-0e729a6ac6e5',
+                                'version': '2020-10-20T15:53:50.322559Z',
+                                'name': '4d6f6c96-2a83-43d8-8fe1-0f53bffd4674.HumanLiver.zip',
+                                'size': 93497178,
+                                'matrix_cell_count': None,
+                                'source': 'Contributor',
+                                'strata': 'genusSpecies=Mus musculus;'
+                                          'developmentStage=adult;'
+                                          'organ=liver;'
+                                          'libraryConstructionApproach=10X v2 sequencing'
+                            }
+                        ]
+                    }
+                ]
+            },
+            '90bf705c-d891-5ce2-aa54-094488b445c6': {
+                'matrices': [],
+                'contributor_matrices': [
+                    {
+                        'file': [
+                            {
+                                # Analysis file. The 'strata' value was gathered by walking
+                                # the project graph. Source from file_source.
+                                'uuid': '2ecedea4-b90c-5025-9967-fc08f27a4dc6',
+                                'version': '2021-01-01T00:00:00.000000Z',
+                                'name': 'AP1_file.h5ad',
+                                'size': 143,
+                                'matrix_cell_count': 2100,
+                                'source': 'Contributor',
+                                'strata': 'genusSpecies=Homo sapiens;'
+                                          'developmentStage=human adult stage;'
+                                          'organ=blood;'
+                                          'libraryConstructionApproach=10x v3 sequencing'
+                            }
+                        ]
+                    }
+                ]
+            }
         }
 
         num_docs = Counter()
+        matrices = {}
         for hit in hits:
             entity_type, aggregate = self._parse_index_name(hit)
             num_docs[entity_type, aggregate] += 1
             if entity_type == 'projects' and aggregate:
-                contents = hit['_source']['contents']
-                self.assertEqual(expected_matrices, one(contents['matrices']))
-                self.assertEqual(expected_contributor_matrices, one(contents['contributor_matrices']))
+                project_id = hit['_source']['entity_id']
+                assert project_id not in matrices, project_id
+                matrices[project_id] = {
```

Comment:
Write this as a dict comprehension over the keys. I will be two lines but will better express the correlation of the keys in  expectations and actuals.

Reply:
```diff
Index: test/indexer/test_hca_indexer.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/test/indexer/test_hca_indexer.py b/test/indexer/test_hca_indexer.py
--- a/test/indexer/test_hca_indexer.py	(revision f0a59e72001ce44c85206331d4b808249c9d65d1)
+++ b/test/indexer/test_hca_indexer.py	(date 1623174363230)
@@ -584,9 +584,8 @@
                 project_id = hit['_source']['entity_id']
                 assert project_id not in matrices, project_id
                 matrices[project_id] = {
-                    k: v
-                    for k, v in hit['_source']['contents'].items()
-                    if k in ('matrices', 'contributor_matrices')
+                    k: hit['_source']['contents'][k]
+                    for k in ('matrices', 'contributor_matrices')
                 }
         self.assertEqual(expected_matrices, matrices)
         expected_num_docs = {
```